sidekick_core/lib/sidekick_core.dart

```diff
@@ -134,15 +135,77 @@ class SidekickCommandRunner<T> extends CommandRunner<T> {
     exitCode = 0;
 
     final unmount = mount();
+
+    String? command;
     try {
-      final result = await super.run(args);
+      final parsedArgs = parse(args);
+      command = parsedArgs.command?.name;
+      final result = await super.runCommand(parsedArgs);
       return result;
     } finally {
+      if (_isUpdateCheckEnabled && command != 'update') {
```

Comment:
what if another command has a subcommand named `update`. Will this run or not?

Reply:
good catch, the check is now more accurate and I added some tests to verify this